e2e/relayer/relayer.go

```diff
@@ -51,6 +57,87 @@ func New(t *testing.T, cfg Config, logger *zap.Logger, dockerClient *dockerclien
 	}
 }
 
+// ApplyPacketFilter applies a packet filter to the hermes config file, which specifies a complete set of channels
+// to watch for packets.
+func ApplyPacketFilter(ctx context.Context, t *testing.T, r ibc.Relayer, chainID string, channels []ibc.ChannelOutput) error {
+	t.Helper()
+
+	h, ok := r.(*hermes.Relayer)
+	if !ok {
+		t.Logf("relayer %T does not support packet filtering, or it has not been implemented yet.", r)
+		return nil
+	}
+
+	return modifyHermesConfigFile(ctx, h, func(config map[string]interface{}) error {
+		chains, ok := config["chains"].([]map[string]interface{})
+		if !ok {
+			return fmt.Errorf("failed to get chains from hermes config")
+		}
+		var chain map[string]interface{}
+		for _, c := range chains {
+			if c["id"] == chainID {
+				chain = c
+				break
+			}
+		}
+
+		if chain == nil {
+			return fmt.Errorf("failed to find chain with id %s", chainID)
+		}
+
+		var channelIDs [][]string
+		for _, c := range channels {
+			channelIDs = append(channelIDs, []string{c.PortID, c.ChannelID})
+		}
+
+		// [chains.packet_filter]
+		//	# policy = 'allow'
+		//	# list = [
+		//	#   ['ica*', '*'],
+		//	#   ['transfer', 'channel-0'],
+		//	# ]
+
+		// TODO(chatton): explicitly enable watching of ICA channels
+		// this will ensure the ICA tests pass, but this will need to be modified to make sure
+		// ICA tests will succeed in parallel.
```

Comment:
I won't create another issue for this since it will be implicitly required to fix before our tests will pass in parallel